### ISSUE TITLE
Numeric label missing on Gauge card

### ISSUE THREAD

**Issue by prilly-dev** [home-assistant/frontend#11956]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

See this screenshot

![Screenshot_20220306-070653_Home Assistant](https://user-images.githubusercontent.com/66376450/156911581-debcaa0d-7b02-40f4-813f-26f990a26263.jpg)

I have issues with the data is not loaded consistanly, both for browser view and in app. It load data about 80% of the time, if i refresh the data load correctly.

### Describe the behavior you expected

Load data

### Steps to reproduce the issue

1.
2.
3.
...

### What version of Home Assistant Core has the issue?

2022 +

### What was the last working version of Home Assistant Core?

no idea, this has been a problem for atlest a year

### Which operating system are you using to run this browser?

windows, linux, android and ios

**Comment by spacegaier** [comment]
With missing data you mean the state value in the gauge cards (temperature and percentage in your screenshot)? Is that the only UI element where this happens? Any errors in the browser console?

**Comment by prilly-dev** [comment]
Yes the data is missing in the card on 20% of the page loadings,  yes this is the only place it happens, no errormessage or anything. It happens on all browsers and the app for android and ios.

But it only happens on this two lines, the page has more gauges further down and it never happens on those.

**Comment by prilly-dev** [comment]
This issue still exist in latest version

**Comment by prilly-dev** [comment]
Issue is still a problem

**Comment by prilly-dev** [comment]
This is still a issue

**Comment by prilly-dev** [comment]
I will be recreating this bugreport

**Comment by RubenKelevra** [comment]
Can confirm this bug for 2025.2.0b6: It will not show the numbers below the gauges, but the gauges itself show the data correctly, as it seems.

The numbers will appear as soon as an updated value is pushed to the UI, that's why two values are visible here in the screenshot - after loading the page didn't showed them and then they appeared.


This also never happens if you do a fresh start of the UI, it's more an issue if you tab back to the app / browser tab after it's been idle and then switch to a different view in the UI, and it's kinda rare to happen. Maybe 1-2% of the time. But it's been consistently an issue as far as I remember with Firefox, Chrome and the Android app.

![Image](https://github.com/user-attachments/assets/943eee30-14dc-4eb0-a4dd-f18b63c25739)

**Comment by rvanmaanen** [comment]
I can (and unfortunately do) reproduce this behavior consistently. 

Below is a part of a dashboard that is causing the issue. I removed 2 cards from the grid that increase/decrease volume. The entire grid is not visible by default, but only turns visible when 'sensor.remote_current_app_active' turns 'True'. When it does, every time the number of the gauge is missing, until I:
 
* On the browser: refresh or change volume
* On the Android app: move away from the dashboard and come back, or change volume

When I do either of these things, it has no impact on the other. So getting the value visible on the browser doesn't change a thing for the app. The sensor.receiver_volume is never Unknown or Unavailable and looking at the developer tools it also really shows a value. For completeness sake I'll also post the yaml of the sensor below.

What does NOT help in getting the numeric value to show:

* Updating the entities of my receiver or the receiver volume sensor
* Waiting

**Dashboard YAML:**

`square: false
type: grid
cards:
  - square: true
    type: grid
    columns: 3
    cards:
      - type: gauge
        entity: sensor.receiver_volume
        needle: false
        min: 0
        max: 100
        tap_action:
          action: perform-action
          perform_action: script.mute_unmute_receiver_volume
          target: {}
        hold_action:
          action: perform-action
          perform_action: media_player.volume_set
          target:
            entity_id: media_player.denon_avr_x2800h
          data:
            volume_level: 0.5
        name: Mute / Unmute
        severity:
          green: 40
          yellow: 60
          red: 70
columns: 1
visibility:
  - condition: state
    entity: sensor.remote_current_app_active
    state: "True"
`

**Sensor YAML:**
`{% if states('media_player.denon_avr_x2800h') == 'off' %}
0
{% elif is_state_attr('media_player.denon_avr_x2800h', 'is_volume_muted', True) %}
0
{% elif state_attr('media_player.denon_avr_x2800h', 'volume_level') is number and state_attr('media_player.denon_avr_x2800h', 'volume_level') > 0 %}
{{(state_attr('media_player.denon_avr_x2800h', 'volume_level')*100)|int}}
{% else %}
0
{% endif %}
`

**Comment by RubenKelevra** [comment]
Changing volume is the weirdest workaround I've ever heard 😂

Next time it's happening I'll try that.

**Comment by rvanmaanen** [comment]
Noo that's only in my case as my gauge is showing the volume. It's like you said: "The numbers will appear as soon as an updated value is pushed to the UI"

**Comment by RubenKelevra** [comment]
Ahhh! That makes sense. Guess I need a coffee.

Yeah, as soon as an update of the values are pushed, the text appears

**Comment by RubenKelevra** [comment]
Can confirm it for 2025.2.4:

![Image](https://github.com/user-attachments/assets/00587282-9aa6-473f-822f-260c712e49f2)

**Comment by karwosts** [comment]
I've seen this too a few times, the value is present in the HTML, but I see the viewBox attribute is set to `0 0 0 0` in cases where it failed. 

I guess something goes wrong with rescaleSvg, but I've never been able to observe in a development environment, so it's a bit hard to pin down.

```
  private _rescaleSvg() {
    // Set the viewbox of the SVG containing the value to perfectly
    // fit the text
    // That way it will auto-scale correctly
    const svgRoot = this.shadowRoot!.querySelector(".text")!;
    const box = svgRoot.querySelector("text")!.getBBox()!;
    svgRoot.setAttribute(
      "viewBox",
      `${box.x} ${box!.y} ${box.width} ${box.height}`
    );
  }
```

**Comment by raqua** [comment]
@RubenKelevra I am experiencing this on fresh open tabs and HA Android app.

**Comment by RubenKelevra** [comment]
@raqua well, this could actually be the same bug. The issue here is the deferred painting of the webpage. Meaning the webpage is not painted until every style sheet element is loaded. If the gauge gets its last update before this moment, the page won't be visible, while the gauge update happened.

I'm not that deep into webpage development, but I think what's missing is that there's a listener for the event "visibilitychange" which will run `_rescaleSvg()` (in `src/components/ha-gauge.ts`), after a check if the document is hidden.

**Comment by supergregg** [comment]
Bump, I also have this issue, primarily in the Android app.

**Comment by karwosts** [comment]
I think I've finally been able to reproduce this, should be able to make a fix.

**Comment by prilly-dev** [comment]
Nice work 👍
